### ISSUE TITLE
Find and hook for a static constructor?

### ISSUE THREAD

**Issue by IgorEisberg** [rovo89/XposedBridge#187]
I don't see this mentioned any where in the documentation, but I wanna hook to a static constructor (aka class initializer).
I just can't find any way to do that in Xposed.
There are both instance constructor (public constructor &lt;init&gt;()V) and class initializer (static constructor &lt;clinit&gt;()V). How should I approach this?

**Comment by rovo89** [comment]
Interesting question. I don't think Java reflection has a way to get a reference to the static constructor. As all the XposedHelpers methods are based upon standard reflection techniques, that's the first hurdle that I think can't be solved. Besides that, you might accidently initialize the class while preparing to hook it. Or it might have been initialized before. Maybe it would be easier to just ensure that it's initialized and then do your modifications. What's the real problem you want to solve?

**Comment by IgorEisberg** [comment]
Hmm I'm trying to add a data type (Opus extension support) to android.media.MediaFile.
Unfortunately this class seems to be static, as in, every method is static and an instance is never created. I did find a workaround as another class exists in my ROM called android.media.MediaFileInjector that has a hookable static addMiuiFileType()V method so that works, but now I got another issue...
Since I got Android 7.0 and my Xposed isn't official there's a bug that package "android" never being intercepted by "handleLoadPackage", it just skips it and goes straight to "com.android.provider.settings" which makes hooking to system classes impossible. Ever seen anything like this?

**Comment by rovo89** [comment]
> Unfortunately this class seems to be static, as in, every method is static and an instance is never created.

Still, you should be able to use the `callStaticMethod()` and `setStaticObjectField()` helpers to manipulate the data. Why does it have to be in the static constructor?

> Since I got Android 7.0 and my Xposed isn't official there's a bug that package "android" never being intercepted by "handleLoadPackage", it just skips it and goes straight to "com.android.provider.settings" which makes hooking to system classes impossible.

I can't confirm that. Did you try something like this?
```java
@Override
public void handleLoadPackage(XC_LoadPackage.LoadPackageParam lpparam) throws Throwable {
	XposedBridge.log(lpparam.packageName);
}
```

But maybe that's a shortcoming in the official versions, just wait for the official one.

**Comment by IgorEisberg** [comment]
Yeah that's exactly what I did. "android" was supposed to be the first logged package, like on my Xperia device (Lollipop). But on Nougat, "android" isn't being logged at all. =.=
